elide-core/src/main/java/com/yahoo/elide/graphql/Environment.java

```diff
@@ -36,19 +38,26 @@
     public Environment(DataFetchingEnvironment environment) {
         Map<String, Object> args = environment.getArguments();
 
-        requestScope = (RequestScope) environment.getContext();
+        requestScope = (GraphQLRequestScope) environment.getContext();
 
         filters = Optional.ofNullable((String) args.get(ModelBuilder.ARGUMENT_FILTER));
-        offset = Optional.ofNullable((String) args.get(ModelBuilder.ARGUMENT_OFFSET));
+        offset = Optional.ofNullable((String) args.get(ModelBuilder.ARGUMENT_AFTER));
         first = Optional.ofNullable((String) args.get(ModelBuilder.ARGUMENT_FIRST));
         sort = Optional.ofNullable((String) args.get(ModelBuilder.ARGUMENT_SORT));
 
         parentType = environment.getParentType();
 
         outputType = environment.getFieldType();
 
-        parentResource = environment.getSource() instanceof RequestScope ? null
-                : (PersistentResource) environment.getSource();
+        rawSource = environment.getSource();
+
+        if (rawSource instanceof PersistentResourceContainer) {
+            parentResource = ((PersistentResourceContainer) rawSource).getPersistentResource();
+        } else if (rawSource instanceof PersistentResource) {
```

Comment:
Will this ever be true anymore?

Reply:
I thought there were certain situations, but maybe not. I'll go through and double check.